k2/python/k2/fsa.py

```diff
@@ -50,60 +92,58 @@ def __init__(self, s: str, openfst: bool = False):
         Args:
           s:
             The input string. Refer to the above comment for its format.
-          openfst:
-            Optional. If true, the string form has the weights as costs,
-            not scores, so we negate as we read.
         '''
-        fsa: _Fsa
-        aux_labels: Optional[torch.Tensor]
-
-        fsa, aux_labels = _fsa_from_str(s, openfst)
-
-        self._fsa = fsa
-        self._aux_labels = aux_labels
+        # Figure out acceptor/transducer for k2 fsa.
+        acceptor = True
+        line = s.strip().split('\n', 1)[0]
+        fields = line.strip().split()
+        assert len(fields) == 4 or len(fields) == 5
+        if len(fields) == 5:
+            acceptor = False
 
-    @classmethod
-    def _create(cls, fsa: _Fsa,
-                aux_labels: Optional[torch.Tensor] = None) -> 'Fsa':
-        '''For internal use only.
-        '''
         ans = cls.__new__(cls)
         super(Fsa, ans).__init__()
+        fsa, aux_labels = _fsa_from_str(s, acceptor, False)
```

Comment:
Instead of having aux_labels as a member directly, we may have it as one of the `per_arc` pseudo-members.
Basically, if we have an Fsa or FsaVec called f, we will have a bunch of fields like f.aux_labels, f.scores, f.foo, f.bar and so on.  They are not actually members but will correspond to keys in some kind of dict.  We'll have fairly general purpose code for propagating them; the main constraint is that the 1st index into them will correspond to the arc-index. 
In the specific case of aux_labels, it may end up getting treated specially for certain operations, like Invert() which would swap those labels with the `symbol` in the arc.

Reply:
I'm not sure if I understand the pseudo-member part. Is this on the Python side, or on the C++ side accessed by Python through pybind? Do you have an example on this?

Reply:
I think `pseudo-member` is via `__getattr__` and `__setattr__`, which is mentioned in one of the issues Dan just created.